modules/programs/htop.nix

```diff
@@ -0,0 +1,331 @@
+    options = {
+      kind = mkOption {
+        type = types.enum (attrNames meters);
+        example = "AllCPUs";
+        description = "What kind of meter";
+      };
+
+      mode = mkOption {
+        type = types.enum [ 1 2 3 4 ];
+        example = 2;
+        description = "Which mode the meter should use, one of 1(Bar) 2(Text) 3(Graph) 4(LED)";
+      };
+    };
+  });
+
+  # Converts the meter to an attribute set if it isn't already
+  meterStandard = m: if builtins.isString m then {
+    kind = m;
+    mode = meters.${m};
+  } else m;
+
+
+  meterType = types.submodule {
+    options = {
+      left = mkOption {
+        description = "Left meters";
+        default = [ "AllCPUs" "Memory" "Swap" ];
+        example = [
+          "Memory"
+          "LeftCPUs2"
+          "RightCPUs2"
+          { kind = "CPU"; mode = 3; }
+        ];
+        type = types.listOf singleMeterType;
+      };
+      right = mkOption {
+        description = "Right meters";
+        default = [ "Tasks" "LoadAverage" "Uptime" ];
+        example = [
+          { kind = "Clock"; mode = 4; }
+          "Uptime"
+          "Tasks"
+        ];
+        type = types.listOf singleMeterType;
+      };
+    };
+  };
+in
+{
+  options.programs.htop = {
+    enable = mkEnableOption "Enable htop";
+
+    fields = mkOption {
+      type = types.listOf (types.enum (attrNames fields));
+      default = [ "PID" "USER" "PRIORITY" "NICE" "M_SIZE" "M_RESIDENT" "M_SHARE" "STATE" "PERCENT_CPU" "PERCENT_MEM" "TIME" "COMM" ];
+      example = [ "PID" "USER" "PRIORITY" "PERCENT_CPU" "M_RESIDENT" "PERCENT_MEM" "TIME" "COMM" ];
+      description = "Fields";
+    };
+
+    sortKey = mkOption {
+      type = types.enum (attrNames fields);
+      default = "PERCENT_CPU";
+      example = "TIME";
+      description = "Which field to use for sorting";
+    };
+
+    sortDescending = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to sort descending or not";
+    };
+
+    hideThreads = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Hide threads";
+    };
+
+    hideKernelThreads = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Hide kernel threads";
+    };
+
+    hideUserlandThreads = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Hide Userland process threads";
+    };
+
+    shadowOtherUsers = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Shadow other users' processes";
+    };
+
+    showThreadNames = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Show custom thread names";
+    };
+
+    showProgramPath = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Show program path";
+    };
+
+    highlightBaseName = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Highlight program \"basename\"";
```

Comment:
Since the descriptions are in docbook format this should probably be `"Highlight program <quote>basename</quote>."`